### PR TITLE
automatically determine extent if not specified

### LINKED ISSUE #101: Automatically set extent on 1d and 2d histograms to the range of the dimension(s)
It would be really useful to not have to specify the extent on the 1d and 2d histograms - the default could be to just set them to the range (or to some nice value encompassing the range).

### PR DESCRIPTION
**Pull request by tarekrached**
Fixes #101 

"tested" by removing extents from `flights/index.ts` and viewing the histograms.

(Also added some stuff to `.gitignore` for those of us still using npm)

### CODE REVIEW

File: src/db/arrow.ts [hunk 1]
```diff
@@ -407,4 +407,23 @@ export class ArrowDB<V extends string, D extends string>
 
     return cubes;
   }
+
+  public getDimensionExtent(dimension: Dimension<D>): Interval<number> {
```
Comment (domoritz) [inline src/db/arrow.ts]: This needs to be added to the API and the mapd DB as well.

**Comment by tarekrached** [comment]
I added `getDimensionExtent()` to the DataBase interface in `src/db/db.ts`.  We're not using mapd, so I'm not entirely sure how to add it to `src/db/mapd.ts`

File: .gitignore [hunk 2]
```diff
@@ -9,4 +9,8 @@ gaia-mapd/app.scss
 gaia-mapd/index.html
 gaia-mapd/index.ts
 node_modules
-yarn-error.log
+
+npm-debug.log*
+yarn-debug.log*
```
Comment (domoritz) [inline .gitignore]: Please sort the file

File: src/api.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ export interface Dimension<D> {
   title?: string;
 
   /** Initial domain for the dimension. */
-  extent: Interval<number>;
+  extent?: Interval<number>;
```
Comment (domoritz) [inline src/api.ts]: Please add a comment about the optional arg.

File: .gitignore [hunk 4]
```diff
@@ -1,6 +1,7 @@
 .cache
 .rpt2_cache*
 .vscode/tags
+/package-lock.json
```
Comment (domoritz) [inline .gitignore]: / is not necessary

```suggestion
package-lock.json
```

**Pull request merged**